### PR TITLE
Add Dockerfile for Red Hat UBI

### PR DESCRIPTION
**Pull request by spotlesstofu**
Adding a Dockerfile to generate a new image, hopefully good enough for tests and experiments.

Resolves: https://issues.redhat.com/browse/KATA-3029

### CODE REVIEW

**Comment by spotlesstofu** [comment]
To give some credit, #398 made this possible.

**Comment by fitzthum** [comment]
Nice. Can you add a DCO sign-off to your commit message. Otherwise the CI will complain

**Review by fitzthum — APPROVED** [review]
LGTM

Maybe we should add a test yaml to trigger building this. Could be a follow-up, I guess.

**Review by bpradipt — COMMENTED** [review]
/lgtm
Thanks @spotlesstofu

File: kbs/docker/Dockerfile.rhel-ubi [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+# Build.
+WORKDIR /usr/src/kbs
+COPY . .
+ARG KBS_FEATURES=coco-as-builtin,rustls,resource,opa
+RUN \
+cargo install --locked --root /usr/local/ --path kbs/src/kbs --no-default-features --features ${KBS_FEATURES} && \
+# Collect linked files necessary for the binary to run.
+mkdir -p /root/trustee/lib64 && \
+ldd /usr/local/bin/kbs | sed 's@.*\s/@/@' | sed 's/\s.*//' | xargs -I {} cp {} /root/trustee/lib64
```
Comment (mythi) [inline kbs/docker/Dockerfile.rhel-ubi:23]: @spotlesstofu @bpradipt did you test this using TDX also? There are DCAP libs that do not show up using ldd.

Comment (spotlesstofu) [reply kbs/docker/Dockerfile.rhel-ubi:23]: No, I didn't test with TDX. I can update the Dockerfile to keep all the contents of the RPMs, what do you think?

Comment (mythi) [reply kbs/docker/Dockerfile.rhel-ubi:23]: The builder does not install the necessary runtime libraries. The final image is expected to have these: https://github.com/confidential-containers/trustee/blob/0b61fabcd3729bad00cb90a7d2eb5ea031eb4abc/kbs/docker/Dockerfile#L56-L57

**Pull request merged**